src/Indigo/Contracts/Holdings/Types.hs

```diff
@@ -0,0 +1,128 @@
+-- SPDX-FileCopyrightText: 2020 TBD
+--
+-- SPDX-License-Identifier: LicenseRef-Proprietary
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Indigo.Contracts.Holdings.Types
+  ( Parameter (..)
+  , Storage
+  , StorageFields (..)
+  , TokenMeta (..)
+  , dummyMeta
+  , mkStorage
+  )
+where
+
+import Indigo
+
+import Fmt (Buildable(..), (+|), (|+))
+
+import qualified Lorentz.Contracts.ManagedLedger as ML
+import Lorentz.TypeAnns (HasTypeAnn)
+
+type instance ErrorArg "notInTransferAdminRightsMode" = ()
```

Comment:
I'd move this in a separate module (see previous comment about `-Wno-orphans`, it could simply be `Indigo.Contracts.Holdings.Errors`